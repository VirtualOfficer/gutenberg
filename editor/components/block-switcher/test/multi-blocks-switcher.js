/**
 * External dependencies
 */
import Shallow from 'react-test-renderer/shallow';

const shallowRenderer = new Shallow();

/**
 * Internal dependencies
 */
import { MultiBlocksSwitcher } from '../multi-blocks-switcher';

describe( 'MultiBlocksSwitcher', () => {
	test( 'should return null when the selection is not a multi block selection.', () => {
		const isMultiBlockSelection = false;
		const selectedBlockUids = [
			'an-uid',
		];
		shallowRenderer.render(
			<MultiBlocksSwitcher
				isMultiBlockSelection={ isMultiBlockSelection }
				selectedBlockUids={ selectedBlockUids }
			/>
		);

		expect( shallowRenderer.getRenderOutput() ).toBeNull();
	} );

	test( 'should return a BlockSwitcher element matching the snapshot.', () => {
		const isMultiBlockSelection = true;
		const selectedBlockUids = [
			'an-uid',
			'another-uid',
		];
		shallowRenderer.render(
			<MultiBlocksSwitcher
				isMultiBlockSelection={ isMultiBlockSelection }
				selectedBlockUids={ selectedBlockUids }
			/>
		);

		expect( shallowRenderer.getRenderOutput() ).toMatchSnapshot();
	} );
} );
