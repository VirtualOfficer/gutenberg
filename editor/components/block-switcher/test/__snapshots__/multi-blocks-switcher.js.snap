// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiBlocksSwitcher should return a BlockSwitcher element matching the snapshot. 1`] = `
<WithSelect(WithDispatch(WithSettingsComponent))
  key="switcher"
  uids={
    Array [
      "an-uid",
      "another-uid",
    ]
  }
/>
`;
